iceoryx_hoofs/source/posix_wrapper/shared_memory_object/shared_memory.cpp

```diff
@@ -33,73 +33,135 @@ namespace iox
 {
 namespace posix
 {
-// NOLINTNEXTLINE(readability-function-size) todo(iox-#832): make a struct out of arguments
-SharedMemory::SharedMemory(const Name_t& name,
-                           const AccessMode accessMode,
-                           const OpenMode openMode,
-                           const mode_t permissions,
-                           const uint64_t size) noexcept
+static int getOflagsFor(const AccessMode accessMode, const OpenMode openMode) noexcept
 {
-    m_isInitialized = true;
+    int oflags = 0;
+    oflags |= (accessMode == AccessMode::READ_ONLY) ? O_RDONLY : O_RDWR;
+    oflags |= (openMode != OpenMode::OPEN_EXISTING) ? O_CREAT : 0;
+    oflags |= (openMode == OpenMode::EXCLUSIVE_CREATE) ? O_EXCL : 0;
+    return oflags;
+}
+
+cxx::string<SharedMemory::Name_t::capacity() + 1> addLeadingSlash(const SharedMemory::Name_t& name) noexcept
+{
+    cxx::string<SharedMemory::Name_t::capacity() + 1> nameWithLeadingSlash = "/";
+    nameWithLeadingSlash.append(cxx::TruncateToCapacity, name);
```

Comment:
Just wondering if `cxx::TruncateToCapacity` is necessary since the source string is smaller than the destination string. Is the optimization not yet implemented?

Reply:
It is always required since append is pure runtime since we do not know at compile time what kind of content is already present in the string. With `assign` this is possible but not with `append`.

Reply:
Okay, you are right. We have here more information than the compiler. This is nothing for this PR, but would it be possible to have a `constexpr` `cxx::string`? This question is directed more towards @FerdinandSpitzschnueffler

Reply:
@elBoberido To be honest, I'm not sure if it would suffice to make all the constructors `constexpr` (the default c'tor already is), bit I'd like to try and will add a todo to #253